src/date-adapters/luxon/index.ts

```diff
@@ -0,0 +1,187 @@
+import { DateTime } from 'luxon';
+import { DateAdapter } from '../date-adapter';
+
+export function adapterFactory(luxon): DateAdapter {
+  /* istanbul ignore next */
+  function coerceDateTime(date: Date | number): DateTime {
+    return typeof date === 'number'
+      ? luxon.DateTime.fromMillis(date)
+      : luxon.DateTime.fromJSDate(date);
+  }
+
+  function addDays(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ days: amount }).toJSDate();
+  }
+
+  function addHours(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ hours: amount }).toJSDate();
+  }
+
+  function addMinutes(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ minutes: amount }).toJSDate();
+  }
+
+  function addSeconds(date: Date | number, amount: number): Date {
+    return coerceDateTime(date).plus({ seconds: amount }).toJSDate();
+  }
+
+  function differenceInDays(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('days')
+    );
+  }
+
+  function differenceInMinutes(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('minutes')
+    );
+  }
+
+  function differenceInSeconds(
+    dateLeft: Date | number,
+    dateRight: Date | number
+  ): number {
+    return Math.trunc(
+      coerceDateTime(dateLeft).diff(coerceDateTime(dateRight)).as('seconds')
+    );
+  }
+
+  function endOfDay(date: Date | number): Date {
+    return coerceDateTime(date).endOf('day').toJSDate();
+  }
+
+  function endOfMonth(date: Date | number): Date {
+    return coerceDateTime(date).endOf('month').toJSDate();
+  }
+
+  function endOfWeek(
+    date: Date | number,
+    options?: {
+      weekStartsOn?: number;
+    }
+  ): Date {
+    return coerceDateTime(date)
+      .endOf('week', { useLocaleWeeks: true })
+      .plus({ days: /* istanbul ignore next */ options?.weekStartsOn ?? 0 })
+      .toJSDate();
+  }
+
+  function getDay(date: Date | number): number {
+    return coerceDateTime(date).weekday % 7;
+  }
+
+  /* istanbul ignore next */
+  function getMonth(date: Date | number): number {
+    return coerceDateTime(date).month % 12;
```

Comment:
Same question with this one

Reply:
This should be - 1 instead. Luxon uses 1-based indexing for months. I'll fix it and add a comment.

Reply:
Good catch, this is a mistake on my end. Luxon uses 1-based counting for their months, but subtracting 1 is enough here. Sorry for that.